uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar-month.component.js

```diff
@@ -248,9 +244,12 @@ define([
       var toDate = moment(leaveRequest.to_date);
 
       while (pointerDate.isSameOrBefore(toDate)) {
-        days.push(_.find(vm.month.days, function (day) {
-          return day.date === pointerDate.format('YYYY-MM-DD');
-        }));
+        // Ensure that pointerDate is in same month/year that component represents
+        if (pointerDate.month() === vm.month.index && pointerDate.year() === vm.month.year) {
```

Comment:
The comment in the code makes sense, but I am not sure why it was necessary in the context of the ticket, so if you document it in the PR, it will be helpful.